client/packages/system/src/Immunisation/api/hooks/useVaccineCourse.ts

```diff
@@ -191,13 +242,26 @@ const useUpdate = (setErrorMessage: Dispatch<SetStateAction<string>>) => {
         return result;
       }
 
-      if (result.__typename === 'UpdateVaccineCourseError') {
-        setErrorMessage(result.error.description);
-        return;
+      let message: string;
+      switch (result.error.description) {
+        case UpdateVaccineCourseError.DatabaseError:
+          message = t('error.database-error');
+          setErrorMessage(message);
+          throw new Error(
+            `${t('error.unable-to-update-vaccine-course')}: ${message}`
+          );
+        case UpdateVaccineCourseError.RecordProgramCombinationAlreadyExists:
+          message = t('error.name-program-duplicate');
+          setErrorMessage(message);
+          throw new Error(
+            `${t('error.unable-to-update-vaccine-course')}: ${message}`
+          );
+        default:
+          throw new Error(`${t('error.unable-to-update-vaccine-course')}`);
       }
     }
 
-    throw new Error(t('error.unable-to-update-vaccine-course'));
+    throw new Error(`${t('error.unable-to-update-vaccine-course')}`);
```

Comment:
the change isn't required - the result of `t` is a string, so there's no need to make it into a string this way. or am I missing something?